### ISSUE TITLE
Work experience order is different than on LinkedIn profile

### ISSUE THREAD

**Issue by DandyDev** [joshuatz/linkedin-to-jsonresume#27]
I was able to successfully export my linkedin profile to JSON Resume format since #26 was fixed. I have a new problem however. The order of work experience items in the resulting JSON Resume file is different than on my linkedin profile. Given that most templates and editors that can work with JSON Resume files, do not try to order it in any way (by date for example), this is problematic.

My [LinkedIn profile]() results in a json file with the following work experience:
```
  "work": [
    {
      "company": "KLM Royal Dutch Airlines",
      "endDate": "2018-08-31",
      "highlights": [],
      "position": "Lead Data Engineer/Architect/Coach",
      "startDate": "2016-08-31",
      "summary": "...",
      "url": ""
    },
    {
      "company": "2organize",
      "endDate": "2012-10-31",
      "highlights": [],
      "position": "Intern Hadoop/Marketing Intelligence",
      "startDate": "2012-07-31",
      "summary": "...",
      "url": ""
    },
    {
      "company": "KLM Royal Dutch Airlines",
      "endDate": "",
      "highlights": [],
      "position": "Director Engineering & Architecture",
      "startDate": "2018-09-30",
      "summary": "...",
      "url": ""
    },
    {
      "company": "Dandymusic",
      "endDate": "2011-06-30",
      "highlights": [],
      "position": "Owner & Pianist/Composer/Arranger",
      "startDate": "2005-09-30",
      "summary": "...",
      "url": ""
    },
    {
      "company": "Hogeschool Rotterdam",
      "endDate": "2014-03-31",
      "highlights": [],
      "position": "Lecturer Datastructures & Algorithms",
      "startDate": "2013-10-31",
      "summary": "...",
      "url": ""
    },
    {
      "company": "Hogeschool Rotterdam",
      "endDate": "2012-07-31",
      "highlights": [],
      "position": "Lecturer Cryptography 101",
      "startDate": "2012-01-31",
      "summary": "...",
      "url": ""
    },
    {
      "company": "2organize",
      "endDate": "2012-10-31",
      "highlights": [],
      "position": "Software Engineer",
      "startDate": "2012-07-31",
      "summary": "...",
      "url": ""
    },
    {
      "company": "Joy in Coding",
      "endDate": "2018-12-31",
      "highlights": [],
      "position": "Freelance Software Engineer / Data Engineer / Firestarter",
      "startDate": "2016-05-31",
      "summary": "...",
      "url": ""
    },
    {
      "company": "Info.nl",
      "endDate": "2014-05-31",
      "highlights": [],
      "position": "Software Engineer",
      "startDate": "2012-11-30",
      "summary": "...",
      "url": ""
    },
    {
      "company": "Hogeschool Rotterdam",
      "endDate": "2013-12-31",
      "highlights": [],
      "position": "Lecturer Cryptography/Security",
      "startDate": "2013-09-30",
      "summary": "...",
      "url": ""
    },
    {
      "company": "ChargeGrid",
      "endDate": "2018-01-31",
      "highlights": [],
      "position": "Co-Founder",
      "startDate": "2016-01-31",
      "summary": "...",
      "url": ""
    },
    {
      "company": "The New Motion",
      "endDate": "2015-11-30",
      "highlights": [],
      "position": "Data Engineer/Scala Engineer",
      "startDate": "2014-06-30",
      "summary": "...",
      "url": ""
    },
    {
      "company": "Digital Airline Operations",
      "endDate": "",
      "highlights": [],
      "position": "Director Engineering & Architecture",
      "startDate": "2018-06-30",
      "url": ""
    },
    {
      "company": "MOvE",
      "endDate": "2011-09-30",
      "highlights": [],
      "position": "Software Engineer",
      "startDate": "2010-06-30",
      "summary": "...",
      "url": ""
    },
    {
      "company": "VPRO",
      "endDate": "2016-08-31",
      "highlights": [],
      "position": "Software Engineer",
      "startDate": "2016-05-31",
      "summary": "...",
      "url": ""
    },
    {
      "company": "MOvE",
      "endDate": "2012-06-30",
      "highlights": [],
      "position": "Lead Engineer",
      "startDate": "2011-09-30",
      "summary": "...",
      "url": ""
    },
    {
      "company": "Universit",
      "endDate": "2016-05-31",
      "highlights": [],
      "position": "Firestarter and Software Engineer",
      "startDate": "2015-12-31",
      "summary": "...",
      "url": ""
    }
  ],
```

**Comment by joshuatz** [comment]
I can see what you mean, and this isn't how the extension used to function; the work section in the output used to mirror the profile webpage exactly, including the same order. However, due to #22, and related issues around "truncated" data, I've had to make some substantial changes that (must) have led to this.

I'll do some digging, and I think I already have an idea of where in my code this probably needs to get fixed.

**Comment by DandyDev** [comment]
Thanks for looking into this!

**Comment by joshuatz** [comment]
@DandyDev Thanks for reporting and the details.

Fix has been released - [v1.4.2](https://github.com/joshuatz/linkedin-to-jsonresume/releases/tag/v1.4.2).

**Comment by DandyDev** [comment]
Thanks for this! Just tested it out and it works as expected and desired!
